### ISSUE TITLE
TLS Client certificate

### ISSUE THREAD

**Issue by tigran-a-secu** [holodeck-b2b/Holodeck-B2B#155]
Hi, 

Question. We're trying to run holodeckb2b 7.0 against an https endnpoint that requires a client certificate. When a message is being sent, we see the error: 
```
2024-11-05 07:30:52,584 (hb2b-core-6)[ERROR] org.holodeckb2b.msgproc.core.REQUEST_OUT_FLOW.CatchFaults - An error occurred while processing messages! Error stack=
        AxisFault : certificate_required(116)
                Caused by: TlsFatalAlertReceived : certificate_required(116)
```
Can holodeckb2b provide a client certificate on such requests? If yes, what key would it use? It seems that there is the only keystore in certmanager configuration: <PrivateKeys> 

Thank you

**Comment by sfieten** [comment]
Yes, Holodeck B2B can supply client TLS certificates. You need to configure the default Java network settings in the start script to provide the key (and if needed trust) store that should be used for TLS.

You can do so by adding the following line to the start script, just before HB2B is started:

`JAVA_OPTS="$JAVA_OPTS -Djavax.net.ssl.keyStore=${HB2B_HOME}/repository/certs/tlskey.p12 -Djavax.net.ssl.keyStorePassword=ExamplePwd"`
